### PR TITLE
add insertcols

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2972

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan - CI passes here cleanly (apart from Out of Memory issues)

File: src/abstractdataframe/abstractdataframe.jl [hunk 1]
```diff
@@ -2544,3 +2544,248 @@ Random.shuffle!(df::AbstractDataFrame) =
...
     permute!(df, randperm(r, nrow(df)))
+
+"""
+    insertcols(df::AbstractDataFrame[, col], (name=>val)::Pair...;
+               after::Bool=false, makeunique::Bool=false, copycols::Bool=true)
+
+Insert a column into a copy of `df` data frame using the [`insertcols!`](@ref)
+function and return the newly created data frame.
+
+See [`insertcols!`](@ref) for the explanation of arguments and examples.
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:2555]: Copy the rest of the docstring here? That's more convenient and we do that e.g. for `sort`/`sort!`.

**Review by nalimilan — COMMENTED** [review]
This PR just moves the `insertcols!` code and docstring, right?

File: src/abstractdataframe/abstractdataframe.jl (hunk 1 above)
Comment (bkamins) [reply src/abstractdataframe/abstractdataframe.jl:2555]: OK - done.

**Comment by bkamins** [comment]
> This PR just moves the `insertcols!` code and docstring, right?

Yes - I do not make any changes in `insertcols!`

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**